plz-media-type.el

```diff
@@ -705,18 +703,18 @@ not.
                              :then (if (symbolp then)
                                        then
                                      (lambda (_)
-                                       (setq buffer (current-buffer))
                                        (when (or (functionp then) (symbolp then))
                                          (funcall then (plz-media-type-then
                                                         plz-media-type--current
-                                                        plz-media-type--response))))))))
-              (cond ((bufferp result)
-                     (plz-media-type--handle-sync-response result))
-                    ((processp result)
-                     result)
-                    (t (user-error "Unexpected response: %s" result))))
-          ;; TODO: How to kill the buffer for sync requests that raise an error?
-          (plz-error (plz-media-type--handle-sync-error error media-types))))
+                                                        plz-media-type--response)))))))
+                   (buffer (process-buffer result)))
```

Comment:
Unfortunatly this won't work here (as seen by the failing tests), because depending on the case, sync or async, `result` can be a buffer (sync) or a process (async). Can you come up with a different solution?

Reply:
Ah, sorry about that. I completely forgot about the `'sync` mode in this case. Will fix.